src/isa/asm.rs

```diff
@@ -1223,3 +1224,26 @@ macro_rules! _int_flags {
         panic!("wrong integer operation flags")
     };
 }
+
+#[cfg(test)]
+mod tests {
+    use core::str::FromStr;
+    use amplify::num::apfloat::ieee;
+    use crate::isa::Instr;
+    use crate::library::Lib;
+    use crate::{Prog, Vm};
+    use paste::paste;
+
+    #[test]
+    fn aluasm_float() {
+        let code = aluasm! {
+            put 1.25,f32[8];
+            ret;
+        };
+        let lib = Lib::assemble(&code).unwrap();
+        let program = Prog::<Instr>::new(lib);
+        let mut vm = Vm::<Instr>::new();
+        vm.run(&program, &());
+        assert_eq!(vm.registers.f32[7].unwrap(), ieee::Single::from_str("1.25").unwrap());
```

Comment:
f32[8] in macro while actual value stored in f32[7]. Is this intentional?

Reply:
the root causes are https://github.com/6293/rust-aluvm/blob/97bd8e7c3e95093a9bf5960b82775f39df3ebeed/src/reg/indexes.rs#L63 and https://github.com/6293/rust-aluvm/blob/8082e98f780c7f29b6c47d07075a21645d998fac/src/isa/asm.rs#L1138

Reply:
we can either
- introduce 0-index everywhere, and rename Reg1 to Reg0
- leave Reg32 1-indexed, just increment the index in macro

I am not sure why the name is 1-indexed, so I am inclined toward the first idea

Reply:
Names are [1]-indexed since it is how the natural numbers have to be used for counting. [0]-indexes in programming is just a mathematical noncence.

But I agree that it is too used by everyone such that it ends up introducing more bugs. I am fine with changing everything back to 0-indexes.

Reply:
will do that in other PR

Reply:
nvm done